### ISSUE TITLE
Placing ice block on button might cause game to get stuck in infinite loop

### ISSUE THREAD

**Issue by Wuzzy2** [wrightmat/zbom#100]
In a dungon in which there is a button that opens a door only as long a block rests on it, it is possible to make the game stuck in an infinite loop. It has to be a button that expects a regular block from the dungeon.

To do this, simply place an ice block with the Cane of Idontknow on such a button. Now the following will happen:

- Door opens
- Camera scrolls to door
- Camera returns to Link
- Camera scrolls to door
- Camera returns to Link
- (repeat infinitely many times)

While this happens, you lose control over Link and there is nothing to abort this sequence. You are forced to reset.

I think there are at least 2 dungeons in the game where this is the case. One such dungeon is the one in Lake Hylia.

ZBOM version: 1.4
Solarus version: 1.6.0

**Comment by wrightmat** [comment]
Unable to recreate

**Comment by Wuzzy2** [comment]
Yeah after 6 years it’s possible the bug has already been fixed and nobody noticed. 😄
